src/user/MOM_wave_interface.F90

```diff
@@ -333,7 +391,7 @@ subroutine MOM_wave_interface_init(time, G, GV, US, param_file, CS, diag )
                  "or the model will fail.", default=1)
       allocate( CS%WaveNum_Cen(CS%NumBands), source=0.0 )
       allocate( CS%STKx0(G%isdB:G%iedB,G%jsd:G%jed,CS%NumBands), source=0.0 )
-      allocate( CS%STKy0(G%isdB:G%iedB,G%jsd:G%jed,CS%NumBands), source=0.0 )
+      allocate( CS%STKy0(G%isd:G%ied,G%jsdB:G%jedB,CS%NumBands), source=0.0 )
```

Comment:
@jiandewang and @alperaltuntas: I am wondering if this bugfix could explain the change in answers. Jiande, can you please re-introduce the bug on the candidate you are evaluating and re-run the fully-coupled test?

Reply:
set the three wave related new parameters to T still got different answer, I will try @gustavo-marques new suggestions

Reply:
@gustavo-marques re-introduce the bug in MOM_wave_interface.F90 still gave me different answer

Reply:
I believe that this specific conversation has been successfully resolved in the main conversation.